src/Entity/DeliveryTimeInformation.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Ups\Entity;
+
+use DOMDocument;
+use DOMElement;
+use Ups\NodeInterface;
+
+/**
+ * @author mazzarito
+ */
+class DeliveryTimeInformation implements NodeInterface
+{
+    const PBT_DOCUMENT_ONLY = '02';
+    const PBT_NON_DOCUMENT = '03';
+    const PBT_PALLET = '04';
+
+    /*
+     * @var string
+     */
+    private $packageBillType;
+
+    /*
+     * @var Pickup
+     */
+    private $pickup;
+
+    /**
+     * @param null|DOMDocument $document
+     *
+     * @return DOMElement
+     */
+    public function toNode(DOMDocument $document = null)
+    {
+        if (null === $document) {
+            $document = new DOMDocument();
+        }
+
+        $node = $document->createElement('DeliveryTimeInformation');
+        $node->appendChild($document->createElement('PackageBillType', $this->getPackageBillType()));
+
+        if ($this->getPickup() !== null) {
+            $node->appendChild($document->createElement('Pickup', $this->getPickup()->toNode()));
```

Comment:
Could you delegate the creation of 'Pickup' to the Pickup object? Looking at NodeInterface (and e.g. Service.php as example) you can send $document as parameter to toNode and do all the generation in there. It offloads all generation to the responsible object.

So something like this:

````
$node->appendChild($this->getPickup()->toNode());
````

And inside Pickup:

````
$node = $document->createElement('Pickup');
// In here adding of elements
````